### PR TITLE
Transformer code for DisplayNames component

### PR DESCRIPTION
**Pull request by snktd**
Tested with '> cargo test --features experimental'

### CODE REVIEW

File: experimental/displaynames/src/provider.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+use icu_provider::prelude::*;
+use tinystr::TinyAsciiStr;
+use zerovec::ZeroMap;
+
+// We use raw TinyAsciiStrs for map keys, as we then don't have to
+// validate them as subtags on deserialization. Map lookup can be
+// done even if they are not valid tags (an invalid key will just
+// become inaccessible).
+type UnvalidatedRegion = TinyAsciiStr<14>;
```
Comment (sffc) [inline experimental/displaynames/src/provider.rs:20]: Should be 3, and drop the -alt-variant ones

See #1225

File: provider/datagen/src/transform/cldr/displaynames/mod.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(DisplayNamesV1::from(data))),
+        })
+    }
+}
+
+impl IterableDataProvider<DisplayNamesV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        Ok(vec![Default::default()])
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/displaynames/mod.rs:32]: This should return all of the locales that are supported in the CLDR input. See the decimal, time zone, etc., transformers for examples of how to impl this function.

File: experimental/displaynames/src/provider.rs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_displaynames::provider),
+)]
+#[yoke(prove_covariance_manually)]
+/// DisplayNames provides mapping between a region code and locale display name.
+pub struct DisplayNamesV1<'data> {
```
Comment (sffc) [inline experimental/displaynames/src/provider.rs:32]: This should be a region-specific struct named, e.g., `TerritoryDisplayNamesV1`

File: experimental/displaynames/src/provider.rs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    /// Mapping for region to locale display name.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub locale_display_names: ZeroMap<'data, UnvalidatedRegion, str>,
```
Comment (sffc) [inline experimental/displaynames/src/provider.rs:35]:
```suggestion
    pub names: ZeroMap<'data, UnvalidatedRegion, str>,
```

File: provider/datagen/src/transform/cldr/displaynames/mod.rs (hunk 2 above)
Comment (snktd) [reply provider/datagen/src/transform/cldr/displaynames/mod.rs]: Done.

File: experimental/displaynames/src/provider.rs (hunk 3 above)
Comment (snktd) [reply experimental/displaynames/src/provider.rs]: Done

File: experimental/displaynames/src/provider.rs (hunk 4 above)
Comment (snktd) [reply experimental/displaynames/src/provider.rs]: Done

File: experimental/displaynames/src/provider.rs (hunk 1 above)
Comment (snktd) [reply experimental/displaynames/src/provider.rs]: I have changed this to 3. But the serde struct for CLDR json files still has TinyAsciiStr<14>. I am excluding "alt-variant" while transforming to provider data struct. 
I think this should work, but I considered few other options too:

1) Using serde(skip_serialization_if) in source data struct to remove "alt-variant". But I read that this is problematic.
2) Removing the "alt-variant" from CLDR territories.json files manually. 

Let me know what you think.

Comment (sffc) [reply experimental/displaynames/src/provider.rs]: The cldr_serde version can have a String, and then you convert it to a TinyAsciiStr in the transform function, skipping entries that don't fit.

Comment (snktd) [reply experimental/displaynames/src/provider.rs]: Done!

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
@snktd Please merge in main and resolve the conflicts.

**Comment by snktd** [comment]
@sffc I somehow messed up while squashing multiple Git commits to one and messed up this PR. I created another PR at https://github.com/unicode-org/icu4x/pull/2635. Sorry for this.

**Pull request closed without merging**